packages/fxa-auth-server/lib/routes/account.js

```diff
@@ -1219,6 +1251,19 @@ module.exports = (log, db, mailer, Password, config, customs, signinUtils, push,
                 return emailRecord;
               });
           })
+          .then(async (emailRecord) => {
+            // Cancel all subscriptions before deletion, if any exist
+            // Subscription records will be deleted from DB as part of account
+            // deletion, but we have to trigger cancellation in payment systems
+            const uid = emailRecord.uid;
+            const subscriptions = await db.fetchAccountSubscriptions(uid) || [];
+            for (const subscription of subscriptions) {
+              const { subscriptionId } = subscription;
+              await subhub.cancelSubscription(uid, subscriptionId);
```

Comment:
We might be able to do a retry for the delete here. We can't re-subscribe the user with subhub, since that requires a freshly-authorized payment token from the Stripe widget (and thus a payment). 

This might be another place where we want to lean on events processing from subhub? I'm assuming that there should be a subscription-cancelled event that comes through, both for when it's cancelled via direct user action here and when cancelled via automatic expiration.

Reply:
Another question I guess is - do we have anywhere else that we do retries or otherwise try to recover from failures in the middle of some non-atomic DB API calls? Looking for an example I could steal

Reply:
Actually, going to resolve this one, since I tweaked the subscription cancelling to happen after password check. The failure case stuff is mentioned in another comment.